code_organization/README.md

```diff
@@ -0,0 +1,229 @@
+
+## Ruby on Rails
+
+The structure of a Ruby on Rails project is defined by the framework itself, as outlined [here](http://guides.rubyonrails.org/getting_started.html). Red Nova Labs adheres to this predefined structure with some exceptions. This section of the guide is focused more on how certain types of files within a Rails project should be formatted.
+
+### Controller File Organization
+
+```ruby
+class PersonsController < ApplicationController
+  # Includes
+  include Authenticatable
+
+  # Layout
+  layout 'facility'
+
+  # Macros
+  supports_pjax
+  cache_sweeper :root_sweeper
+  protect_from_forgery with: :exception
+
+  # Actions
+  before_action :authenticate_user
+
+  # GET /
+  def index
+  end
+
+  # POST /
+  def create
+  end
+
+  # GET /new
+  def new
+  end
+
+  # GET /:id/edit
+  def edit
+  end
+
+  # GET /:id
+  def show
+  end
+
+  # PUT /:id
+  # PATCH /:id
+  def update
+  end
+
+  # DELETE /:id
+  def destroy
+  end
+
+  # Protected and Private Methods
+
+  protected
+
+  def some_protected_method
+  end
+
+  private
+
+  def some_private_method
+  end
+end
+```
+
+### Model File Organization
+
+A mix of bbatsov's [Ruby](https://github.com/bbatsov/ruby-style-guide#classes--modules) and [Rails](https://github.com/bbatsov/rails-style-guide#macro-style-methods) file organization guides.
+
+```ruby
+class User
+  # Default Scope
+  default_scope { where(active: true) }
+
+  # Extends and Includes
+  extend SomeModule
+  include AnotherModule
+
+  # Inner Classes
+  CustomError = Class.new(StandardError)
+
+  # Constants
+  SOME_CONSTANT = 20
+
+  # Attribute Macros
+  attr_reader :name
+  attr_accessor :formatted_date_of_birth
+
+  # Enums
+  enum gender: { female: 0, male: 1 }
+
+  # Association Macros
+  belongs_to :country
+  has_many :authentications, dependent: :destroy
+
+  # Validation Macros
+  validates :email, presence: true
+  validates :username, presence: true
+  validates :username, uniqueness: { case_sensitive: false }
+  validates :username, format: { with: /\A[A-Za-z][A-Za-z0-9._-]{2,19}\z/ }
+  validates :password, format: { with: /\A\S{8,128}\z/, allow_nil: true }
+
+  # Callbacks
+  before_save :cook
+  before_save :update_username_lower
+
+  # Other Macros
+  devise :database_authenticatable
+
+  # Public Class Methods
+  def self.some_method
+  end
+
+  # Initializer
+  def initialize
+  end
+
+  # Public Instance Methods
+  def some_method
+  end
+
+  # Protected and Private Methods
+
+  protected
+
+  def some_protected_method
+  end
+
+  private
+
+  def some_private_method
+  end
+end
+```
+
+#### Model Test File Organization
+
+A model's test file should mostly mirror the organization present in the model itself. `describe` and `context` blocks should be used instead of comments to provide information about the file's structure and code grouping.
+
+```ruby
+describe User do
+  describe('Default Scope') do
```

Comment:
I'm a believer that scopes should not be tested (at least not a a unit test level), so I would leave this example out.

Reply:
I tend to agree and gets me thinking that there should probably be a testing do's and dont's page in this guide too. @bgibbons0531 said this in a thread in Slack on 01-22-2018:

> Would make it more... friendly... to add tests as a visiting dev too. When I add things to FMS, I figure I should add a test around it (even if super small), look through test code base, and then think to myself maybe it's not that important

Even just a list of things that _must_ be tested before opening a pull request because I was on the same page with him in that I didn't know which tests I should include in my first PR.